### PR TITLE
feat: allow to create private cluster-local service (#30)

### LINKED ISSUE #30: Allow to create a private cluster-local service
By default services deployed through Knative are published to an external IP address, making them public. It is also possible to make a service not available off-cluster (e.g if you're deploying a backend service).
We should allow to do it at creation time, maybe with a checkbox in the dialog.
More infos: https://knative.dev/docs/serving/cluster-local-route/

### PR DESCRIPTION
**Pull request by lstocchi**
it resolves #30 

local-network-cluster

### CODE REVIEW

**Review by evidolob — APPROVED** [review]
LGTM

File: src/main/java/com/redhat/devtools/intellij/knative/utils/YAMLBuilder.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/knative/utils/YAMLBuilder.java:1]: Seems to be an IJ common thing, no ?

Comment (lstocchi) [reply src/main/java/com/redhat/devtools/intellij/knative/utils/YAMLBuilder.java:1]: Yeah it can also be moved. Moving it

Comment (lstocchi) [reply src/main/java/com/redhat/devtools/intellij/knative/utils/YAMLBuilder.java]: Moved with https://github.com/redhat-developer/intellij-common/pull/72

File: src/main/java/com/redhat/devtools/intellij/knative/ui/CreateServiceDialog.java [hunk 2]
```diff
@@ -145,30 +155,50 @@ public void changedUpdate(DocumentEvent e) {
...
+            try {
+                JsonNode updated;
+                if (e.getStateChange() == ItemEvent.SELECTED) {
+                    updated = YAMLHelper.addLabelToResource(yaml, "networking.knative.dev/visibility", "cluster-local");
+                } else {
+                    updated = YAMLHelper.removeLabelFromResource(yaml, "networking.knative.dev/visibility");
+                }
+                updateEditor(updated);
+            } catch (IOException ex) {
+                logger.warn(ex.getLocalizedMessage());
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/knative/ui/CreateServiceDialog.java:176]: Exception is missing

File: src/main/java/com/redhat/devtools/intellij/knative/ui/CreateServiceDialog.java [hunk 3]
```diff
@@ -145,30 +155,50 @@ public void changedUpdate(DocumentEvent e) {
...
             logger.warn(e.getLocalizedMessage());
         }
+    }
 
+    private void updateEditor(JsonNode node) {
+        ApplicationManager.getApplication().runWriteAction(() -> {
+            try {
+                editor.getEditor().getDocument().setText(YAMLHelper.JSONToYAML(node));
+            } catch (IOException e) {
+                logger.warn(e.getLocalizedMessage());
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/knative/ui/CreateServiceDialog.java:199]: Same here

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Maybe worth:
- using IJ common 1.1.0-SNAPSHOT
- caught some log.warn without exception object being passed, but there may be other calls from existing code (or move to a separate issue as the whole code base should be checked)

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**